File: ts/example/src/tests/advanced/anon_func_style_query_user.ts
import {ExampleAnonFuncStyleQuery} from "./anon_func_style_query";


export class ActualAnonFuncStyleQueryUser {
  constructor(private query: ExampleAnonFuncStyleQuery) {}


  doSomething(): string {
    // Do something.

    return this.query();
  }
}

File: swift/example/Sources/VanillaMockExample/Advanced/AnonFuncStyleQueryUser.swift
class ActualAnonFuncStyleQueryUser {
    private let query: ExampleAnonFuncStyleQueryProtocol


    init(query: @escaping ExampleAnonFuncStyleQueryProtocol) {
        self.query = query
    }


    func doSomething() -> String {
        // Do something.

        self.query()
    }
}